test/blackbox-tests/test-cases/too-many-parens/run.t

```diff
@@ -8,25 +8,39 @@ are readable.
   3:   (public_name hello)
   4:   (libraries (lib))
   5: ))
-  These parentheses are no longer necessary with dune, please remove them.
+  Error: Atom expected
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
   [1]
 
   $ dune build --root b
   File "dune", line 4, characters 12-17:
    (libraries (lib)))
               ^^^^^
-  These parentheses are no longer necessary with dune, please remove them.
+  Error: 'select' expected
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
   [1]
 
   $ dune build --root c
   File "dune", line 3, characters 7-14:
    (deps (x y z)))
          ^^^^^^^
-  These parentheses are no longer necessary with dune, please remove them.
+  Error: Unknown constructor x
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
   [1]
 
 Checking that extra long stanzas (over 10 lines) are not printed
   $ dune build --root d
   File "dune", line 3, characters 13-192:
-  These parentheses are no longer necessary with dune, please remove them.
+  Error: 'select' expected
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
+  [1]
+
+When the inner syntax is wrong, do not warn about the parens:
+
+  $ dune build --root e
+  File "dune", line 3, characters 7-15:
+   (deps (glob *)) ; this form doesn't exist
+         ^^^^^^^^
+  Error: Unknown constructor glob
+  Hint: try: in dune files, these parentheses are no longer necessary. Try removing them.
```

Comment:
In this particular case, the hint doesn't point the user in the right direction. It seems hard to fix in general, maybe we should just make the message a bit more generic:

```
Hint: dune files require less parentheses than jbuild files.
If you just converted this file from a jbuild file, try removing these parentheses.
```

Reply:
So, we'd display the "unknown constructor" error and this hint? sounds good to me.

Reply:
yep. This is basically a best effort to help the user fix the problem